src/app/wallet/shared/transaction.model.ts

```diff
@@ -67,18 +67,20 @@ export class Transaction implements Deserializable {
 
   /** Calculates the actual amount that was transfered, including the fee */
   /* todo: fee is not defined in normal receive tx, wut? */
-  public getNetAmount() {
+  public getNetAmount(): number {
     const amount: number = +this.getAmountObject().getAmount();
 
     /* If fee undefined then just return amount */
     if (this.fee === undefined) {
       return amount;
     /* sent */
     } else if (amount < 0) {
-      return amount + this.fee;
+      const net = new Amount(+amount + (+this.fee)).getAmount();
```

Comment:
Can be one return, and next return looks the same?

Reply:
I'll smash the two lines into one, but I prefer to keep the else if structure, I'm not very fond of inline ( ? .. : ..) stuff. Makes readability and documenting it tough :/